### PR TITLE
csvparser-filterx: do not traverse column list 3 times

### PR DESCRIPTION
**Pull request by MrAnno**
Depends on #249

csvparser-filterx: do not traverse column list 3 times

- constructing list
- g_list_length()
- traversing list

This also fixes a possible bug where a name-value pair passed to `filterx_string_new()` may not have been null-terminated.

### CODE REVIEW

**Comment by bazsi** [comment]
:+1:

**Review by bazsi — DISMISSED** [review]

**Review by bazsi — DISMISSED** [review]

**Review by bazsi — APPROVED** [review]

**Comment by alltilla** [comment]
I think this is a good step forward.

Recently we have added a support for passing list/dict literals to function arguments. With that we don't need to evaluate the list over and over again, and we can initialize even a `GPtrArray` with the column names in init time. The next step should be supporting this IMO.

For example see: https://github.com/MrAnno/axosyslog/blob/10376b4a6a158999699bb9ba05ec11ead88a2a6e/lib/filterx/filterx-metrics-labels.c#L328

File: modules/csvparser/filterx-func-parse-csv.c [hunk 1]
```diff
@@ -103,12 +103,61 @@ _init_scanner(FilterXFunctionParseCSV *self, GList *string_delimiters, GList *co
...
+
+static inline gboolean
+_fill_col_object_field(FilterXObject *cols, gint64 index, CSVScanner *scanner, FilterXObject *result)
+{
+  FilterXObject *col = filterx_list_get_subscript(cols, index);
+  const gchar *col_name;
+  gsize col_name_len;
+  filterx_object_extract_string(col, &col_name, &col_name_len);
+
+  FilterXObject *key = filterx_string_new(col_name, col_name_len);
```
Comment (alltilla) [inline modules/csvparser/filterx-func-parse-csv.c:148]: Why don't we just use `col`? It seems like we have recreated the same string. `set_subscript()` support `message_value`s as well, so there is no need to ensure `FilterXString`.

Comment (MrAnno) [reply modules/csvparser/filterx-func-parse-csv.c:148]: I didn't notice about this, I just refactored this part. 
I've removed the unnecessary string recreation.

**Pull request merged**